File: data-exercise/spark/spark-core/src/test/scala/com/mqjd/spark/scala/spark01/HelloSparkRDDTest.scala
package com.mqjd.spark.scala.spark01

import com.mqjd.spark.scala.SparkJobTest
import org.scalatest.funsuite.AnyFunSuite

class HelloSparkRDDTest extends AnyFunSuite with SparkJobTest {
  test("given expect dataset when HelloSparkRDD then success") {
    val targetDir = getTargetFile("spark01")
    deleteDirectory(targetDir)
    val args = Array("--output", targetDir)
    HelloSparkRDD.main(args)
    val actualRdd = spark.textFile(targetDir)
    val expectRdd = spark.textFile(getResourceFile("/spark01.txt"))
    assertRddEquals(actualRdd, expectRdd)
  }
}

File: data-exercise/spark/spark-core/src/test/java/com/mqjd/spark/java/spark01/HelloSparkRDDTest.java
package com.mqjd.spark.java.spark01;

import com.mqjd.spark.java.SparkJobTest;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.junit.Test;

public class HelloSparkRDDTest extends SparkJobTest {

    @Test
    public void given_expect_text_when_HelloSparkRDD_then_success() {
        String targetDir = "target/spark01";
        deleteDirectory(targetDir);
        HelloSparkRDD.main(new String[]{"--output", targetDir});
        JavaSparkContext sc = getSparkContext();
        JavaRDD<String> actual = sc.textFile(targetDir);
        JavaRDD<String> expect = sc.textFile(getResourceFile("/spark01.txt"));
        assertRddEquals(expect, actual);
    }

}